src/thread_task.rs

```diff
@@ -813,869 +813,737 @@ unsafe fn delayed_fg_task(c: *const Rav1dContext, ttd: *mut TaskThreadData) {
+                                if (*next_t).sby == start {
+                                    (*f).frame_thread.next_tile_row[p as usize] = ntr;
                                 }
+                                (*next_t).recon_progress = (*next_t).sby + 1;
+                                insert_task(f, next_t, 0 as c_int);
                             }
-                        }
-                        _ => {
-                            if !prev_t.is_null() {
-                                (*prev_t).next = (*t).next;
-                            } else {
-                                (*f).task_thread.task_head = (*t).next;
+                            found = true;
+                            break 'decoding;
+                        } else if (*t).type_0 as c_uint == RAV1D_TASK_TYPE_CDEF as c_int as c_uint {
+                            let p1_1 = (*f).frame_thread.copy_lpf_progress
+                                [((*t).sby - 1 >> 5) as usize]
+                                .load(Ordering::SeqCst);
+                            if p1_1 as c_uint & (1 as c_uint) << ((*t).sby - 1 & 31) != 0 {
+                                found = true;
+                                break 'decoding;
                             }
-                            if ((*t).next).is_null() {
-                                (*f).task_thread.task_tail = prev_t;
+                        } else {
+                            if (*t).deblock_progress == 0 {
+                                unreachable!();
                             }
-                            if (*t).type_0 as c_uint > RAV1D_TASK_TYPE_INIT_CDF as c_int as c_uint
-                                && ((*f).task_thread.task_head).is_null()
-                            {
-                                (*ttd).cur = ((*ttd).cur).wrapping_add(1);
+                            let p1_2 = (*f).frame_thread.deblock_progress.load(Ordering::SeqCst);
+                            if p1_2 >= (*t).deblock_progress {
+                                ::core::intrinsics::atomic_or_seqcst(
+                                    &mut (*f).task_thread.error,
+                                    (p1_2 == TILE_ERROR) as c_int,
+                                );
+                                found = true;
+                                break 'decoding;
                             }
-                            (*t).next = 0 as *mut Rav1dTask;
-                            ::core::intrinsics::atomic_store_seqcst(
-                                &mut (*ttd).cond_signaled,
-                                1 as c_int,
-                            );
-                            pthread_cond_signal(&mut (*ttd).cond);
-                            pthread_mutex_unlock(&mut (*ttd).lock);
-                            current_block = 8464383504555462953;
-                            break;
                         }
                     }
+                    prev_t = t;
```

Comment:
```suggestion
                    // next:
                    prev_t = t;
```

I think the `next:` comment you had before is still helpful for mapping back to the C.